Emby.Server.Implementations/HttpServer/HttpResultFactory.cs

```diff
@@ -699,36 +700,26 @@ private bool IsNotModified(IRequest requestContext, Guid cacheKey, DateTime? las
 
             var ifModifiedSinceHeader = requestContext.Headers.Get("If-Modified-Since");
 
-            if (!string.IsNullOrEmpty(ifModifiedSinceHeader))
+            if (!string.IsNullOrEmpty(ifModifiedSinceHeader)
+                && DateTime.TryParse(ifModifiedSinceHeader, out DateTime ifModifiedSince)
+                && IsNotModified(ifModifiedSince.ToUniversalTime(), cacheDuration, lastDateModified))
             {
-                DateTime ifModifiedSince;
-
-                if (DateTime.TryParse(ifModifiedSinceHeader, out ifModifiedSince))
-                {
-                    if (IsNotModified(ifModifiedSince.ToUniversalTime(), cacheDuration, lastDateModified))
-                    {
-                        return true;
-                    }
-                }
+                return true;
             }
 
             var ifNoneMatchHeader = requestContext.Headers.Get("If-None-Match");
 
-            var hasCacheKey = !cacheKey.Equals(Guid.Empty);
+            bool hasCacheKey = !cacheKey.Equals(Guid.Empty);
 
             // Validate If-None-Match
-            if ((hasCacheKey || !string.IsNullOrEmpty(ifNoneMatchHeader)))
+            if ((hasCacheKey && !string.IsNullOrEmpty(ifNoneMatchHeader)))
```

Comment:
According to Mozilla docs `If-None-Match` has precedence over `If-Modified-Since`, I wonder if reversed order here is the source of our caching problems.
Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match

Reply:
@Bond-009 Can we swap sections handling headers to be spec-compliant?

Reply:
I will do it tomorrow as soon as I’m behind my desktop.

Reply:
As agreed on matrix let's fix in in separate PR.

Reply:
This change doesn't make sense to me now... maybe I'm just tired and dumb from reading other reviews.
Please explain the logic here :)

Reply:
Read docs better... This change is correct